pennylane/transforms/condition.py

```diff
@@ -0,0 +1,98 @@
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Contains the condition transform.
+"""
+from functools import wraps
+from typing import Type
+
+from pennylane.operation import Operation, AnyWires
+from pennylane.measurements import MeasurementValue
+
+
+class Conditional(Operation):
+    """A Conditional Operation.
+
+    Unless you are a Pennylane plugin developer, **you should NOT directly use this class**,
+    instead, use the :func:`qml.cond <.cond>` function.
+
+    The ``Conditional`` class is a container class that defines an operation
+    that should by applied relative to a single measurement value.
+
+    Support for executing ``Conditional`` operations is device-dependent. If a
+    device doesn't support mid-circuit measurements natively, then the QNode
+    will apply the :func:`defer_measurements` transform.
+
+    Args:
+        expr (MeasurementValue): the measurement outcome value to consider
+        then_op (Operation): the PennyLane operation to apply conditionally
+        do_queue (bool): indicates whether the operator should be
+            recorded when created in a tape context
+        id (str): custom label given to an operator instance,
+            can be useful for some applications where the instance has to be identified
+    """
+
+    num_wires = AnyWires
+
+    def __init__(
+        self,
+        expr: MeasurementValue[bool],
+        then_op: Type[Operation],
+        do_queue=True,
+        id=None,
+    ):
+        self.meas_val = expr
+        self.then_op = then_op
+        super().__init__(wires=then_op.wires, do_queue=do_queue, id=id)
+
+
+def cond(measurement, then_op):
+    """Create an operation that applies a version of the provided operation
+    that is conditioned on a value dependent on quantum measurements.
+
+    Args:
+        measurement (MeasurementDependentValue): The measurement dependent
+            value to consider.
+        then_op (Operation): The PennyLane operation to apply if the condition
+            applies.
+
+    Returns:
+        function: A new function that applies the controlled equivalent of ``operation``. The returned
+        operation takes the same input arguments as ``operation``.
+
+    **Example**
+
+    .. code-block:: python3
+
+        dev = qml.device("default.qubit", wires=3)
+
+        first_par = 0.1
+        sec_par = 0.3
+
+        @qml.qnode(dev)
+        @qml.defer_measurements
+        def qnode():
+            m_0 = qml.measure(0)
+            qml.cond(m_0, qml.RY)(first_par, wires=1)
+
+            m_1 = qml.measure(2)
+            qml.cond(m_0, qml.RZ)(sec_par, wires=1)
+            return qml.expval(qml.PauliZ(1))

```

Comment:
Thanks! I'll remove the explicit use of `@qml.defer_measurements`, but might keep the full-fledged example. We have something similar for `qml.ctrl` too and it can be helpful to provide something that users can execute right away.